### PR TITLE
py-ipython: add 8.5.0 and py-win-unicode-console: new package

### PR DESCRIPTION
**Pull request by manuelakuhn**
https://github.com/ipython/ipython/tree/8.5.0
https://github.com/Drekin/win-unicode-console/tree/0.5

Also fix old dependencies.

Version 3.1.0 and 2.3.1 do not build (also before this PR). What should we do with those?

### CODE REVIEW

**Comment by adamjstewart** [comment]
> Version 3.1.0 and 2.3.1 do not build (also before this PR). What should we do with those?

Deprecate.

File: var/spack/repos/builtin/packages/py-ipython/package.py [hunk 1]
```diff
@@ -35,43 +45,55 @@ class PyIpython(PythonPackage):
...
-    def import_modules(self):
-        modules = super(__class__, self).import_modules
+    def skip_modules(self):
 
         if self.spec.satisfies("@8:"):
-            return modules
+            return []
 
         # 'IPython.kernel' is deprecated and fails to import, leave out of
         # 'import_modules' to ensure that import tests pass.
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-ipython/package.py]: Trying to understand this logic. In IPython 7 and older, IPython.kernel was deprecated, so we skip that import. But in IPython 8 it is no longer deprecated and imports fine? Or it no longer exists so there's no reason to skip it? Maybe we can just always skip it if it's the latter.

Comment (manuelakuhn) [reply var/spack/repos/builtin/packages/py-ipython/package.py]: `IPython.kernel` no longer exists in newer version, that's why the test pass now without skipping of modules. The idea behind the distinction between version was to modify `import_modules` as little as possible.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-ipython/package.py]: Let's just always return `IPython.kernel`, not returning anything when it doesn't exist seems like an unnecessary step. Alternatively, we can only skip it after it becomes deprecated.

Comment (manuelakuhn) [reply var/spack/repos/builtin/packages/py-ipython/package.py]: Done.

**Review by adamjstewart — DISMISSED** [review]

File: var/spack/repos/builtin/packages/py-ipython/package.py [hunk 2]
```diff
@@ -35,43 +45,50 @@ class PyIpython(PythonPackage):
...
+        "py-win-unicode-console@0.5:",
+        when="@5:7.9 ^python@:3.5 platform=windows",
+        type=("build", "run"),
+    )
 
-        return [i for i in modules if not any(map(i.startswith, ignored_imports))]
+    # "IPython.kernel" was deprecated for py-ipython@:7 and fails to import, leave
+    # out of "import_modules" to ensure that import tests pass.
+    # for py-ipython@8: "IPython.kernel" was removed
+    skip_modules = ["IPython.kernel"]
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-ipython/package.py]: Can we move this to the top? Maybe below `git`?

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**